### PR TITLE
[TT-7422] keep base API version unaffected by sort.

### PR DESCRIPTION
**Pull request by jeffy-mathew**
Version listing should return base API as the first element without getting affected during sort.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4628/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3883035741)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4628/merge
Commit: c8c5c0c4cddcae511919c112b5d8a94493aa59b2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3883037033)

File: gateway/versions_handler.go [hunk 1]
```diff
@@ -76,22 +74,25 @@ func (h *VersionsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-
-		versionMetas.Metas = append(versionMetas.Metas, meta)
+		})
 	}
 
 	sort.Slice(versionMetas.Metas, func(i, j int) bool {
 		return versionMetas.Metas[i].VersionName < versionMetas.Metas[j].VersionName
 	})
 
+	if baseAPIMeta.ID != "" {
```
Comment (titpetric) [inline gateway/versions_handler.go:91]: This `if` is redundant, I'm asuming we're going to error out at L46

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/4628/merge
Commit: 9c366f874ad49f941bdf6ce50bb95e4419ff0fdf 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3883220753)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4628/merge
Commit: 9c366f874ad49f941bdf6ce50bb95e4419ff0fdf 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3883220753)

File: gateway/versions_handler.go (hunk 1 above)
Comment (jeffy-mathew) [reply gateway/versions_handler.go:91]: baseAPIMeta is already declared, so appending it without a check would add an entry with empty values

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4628/merge
Commit: 80cdc3e223785daf636ef859b025742aff2a02f4 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3891284520)

**Review by furkansenharputlu — APPROVED** [review]

**Pull request merged**